packages/library-base/src/commonMain/kotlin/io/realm/exceptions/RealmException.kt

```diff
@@ -0,0 +1,15 @@
+package io.realm.exceptions
+
+/**
+ * Top level class for all exceptions that are Realm specific and doesn't have a better
+ * existing platform exception like [IllegalArgumentException] or [IllegalStateException].
+ *
+ * In most cases, a subclass of this type will be thrown. Exactly which, will will be documented by
+ * the individual API entry points.
+ */
+public open class RealmException : RuntimeException {
```

Comment:
We should probably wrap the core exceptions in this on in https://github.com/realm/realm-kotlin/blob/master/packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmUtils.kt#L307 instead

Reply:
Yup, that was the plan. It will happen as part of #745